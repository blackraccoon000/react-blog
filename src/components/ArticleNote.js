import React from "react"
import { DateTime } from "luxon"

const ArticleNote = (props) => {
  return (
    <div className="article-item">
      <figure className="article-item__photo">
        { props.link ?
          <img
            className="article-item__img"
            src={props.link}
            alt="dog"
          />
        :
          <div className="article-item__imgDiv">
            <p className="article-item__imgDiv__p">No Image</p>
          </div>
        }
      </figure>
      <div className="article-item__box">
        <h3 className="article-item__title">{props.title}</h3>
        <p className="article-item__text">{props.note}</p>
        <p className="article-item__date">{DateTime.fromMillis(props.createdAt).toFormat('yyyy LL dd')}</p>
      </div>
    </div>
  )
}

export default ArticleNote