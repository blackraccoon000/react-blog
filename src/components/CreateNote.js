import React from "react"
import { connect } from "react-redux"
import { createBlog, removeBlog, editBlog } from "../actions/blogActions"

class CreateNote extends React.Component {
  constructor(props){
    super(props)
    const id = props.match.params.id && props.match.params.id
    const title = props.findData.title && props.findData.title
    const note = props.findData.note && props.findData.note
    this.state = {
      id,
      title,
      note,
      error:""
    }
    this.remove = this.remove.bind(this)
  }
  titleInput = (e) => {
    const title = e.target.value
    this.setState(()=>({ title }))
  }
  noteInput = (e) => {
    const note = e.target.value
    this.setState(()=>({ note }))
  }
  submit = (e) => {
    e.preventDefault()
    const { id, title, note } = this.state
    if( title==="" || note==="" ) {
      this.setState(()=>({ error: "titleとnoteに入力してください" }))
    } else {
      if(id) { this.props.editBlogD(id, { title, note }) }
      else { this.props.createBlogD({ title, note }) }
      this.props.history.push("/dashboard")
    }
  }
  remove = () => {
    this.props.removeBlogD(this.state.id)
  }
  render() {
    return (
    <div className="content-container">
      <h1>This is Create Note Page</h1>
      {this.state.error && <p className="form__error">{this.state.error}</p>}
        <div>
          <form onSubmit={this.submit}>
          <input type="text"
            className="createNote-title"
            type="text"
            autoFocus
            placeholder="title"
            onChange={this.titleInput}
            value={this.state.title}
            data-testid={"title"}
          />
          <textarea
            className="createNote-textarea"
            placeholder="Add a note"
            onChange={this.noteInput}
            value={this.state.note}
            data-testid={"note"}
          ></textarea>
          <button>save</button>
          {this.state.id && <button onClick={this.remove}>remove</button>}
        </form>
      </div>
    </div>
    )
  }
}

const mapStateToProps = (state,ownProps) =>{
  const findData = state.blogData.find(note => note.id === ownProps.match.params.id)
  console.log(findData)
  return {
    findData: findData ? findData : {
      title:"",
      note:""
    }
  }
}

const mapDispatchToProps = (dispatch) => ({
  createBlogD: (createData) => dispatch(createBlog(createData)),
  editBlogD: (id,updates) => dispatch(editBlog(id,updates)),
  removeBlogD: (id) => dispatch(removeBlog(id)),
})

export default connect(mapStateToProps, mapDispatchToProps)(CreateNote)
export { CreateNote }