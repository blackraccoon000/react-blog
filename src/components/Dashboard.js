import React from "react"

const Dashboard = (props) => {
  const history = props.history
  return (
    <div>
      <h1>This Component is Dashboard</h1>
      <button onClick={()=>{
        history.push("/")
        }}>Go Home</button>
    </div>
  )
}

export default Dashboard