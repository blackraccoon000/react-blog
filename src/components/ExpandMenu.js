import React from 'react';
import { Link } from 'react-router-dom';

class Translation extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      flag: false,
    };
  }
  onToggle = () => {
    this.setState({ flag: !this.state.flag });
  };
  render() {
    return (
      <div>
        <div id="cont">
          <div id="menu-fixed">
            <div class="notes">
              <p>Expand the menu!</p>
            </div>
            <a href="#cont">
              <i class="material-icons back">&#xE314;</i>
            </a>
            <a href="#menu-fixed">
              <div class="logo">
                <span></span>
                <p>TITLE</p>
              </div>
              <p class="pmenu">MENU</p>
            </a>
            <ul class="menu">
              <li>
                <i class="material-icons">&#xE88A;</i>
                <p>Home</p>
              </li>
              <li>
                <i class="material-icons">&#xE87A;</i>
                <p>Explore</p>
              </li>
              <li>
                <i class="material-icons">&#xE8CC;</i>
                <p>Shop</p>
              </li>
              <li>
                <i class="material-icons">&#xE8B8;</i>
                <p>Settings</p>
              </li>
              <li>
                <i class="material-icons">&#xE8B6;</i>
                <p>Search</p>
              </li>
            </ul>
            <i class="material-icons info">&#xE88E;</i>
          </div>
        </div>
        <div id="page">
          <div></div>
        </div>
      </div>
    );
  }
}

export default Translation;
