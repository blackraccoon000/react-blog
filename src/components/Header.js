import React from "react"
import NavMenu from "./NavMenu"
import { connect } from "react-redux"
import NoteListFilters from "./NoteListFilters"
class Header extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      flag:false
    }
  }
  onClickBtn = () => {
    this.setState({flag: !this.state.flag })
  }
  render() {
    return (
      <div>
        <header className="header">
          <div className="content-container">
            <h1 className="header__title">My Blog</h1>
            { this.state.flag && <NavMenu history={history}/> }
            { this.state.flag && <NoteListFilters/> }
            <button className="navBtn" data-testid="navBarFlagButton" onClick={this.onClickBtn}>
              {this.state.flag ? "Close" : "Open" }
            </button>
          </div>
        </header>
        <div className="header-alert">
          <div className="content-container">
            { this.props.isAuthenticated && "sign in now" }
          </div>
        </div>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  isAuthenticated: !!state.auth.uid
})

export { Header }
export default connect( mapStateToProps )( Header )