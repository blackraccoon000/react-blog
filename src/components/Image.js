import React from 'react';
import axios from 'axios';
import Spinner from '../img/spinner.png';

class Image extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      images: [],
    };
  }
  componentDidMount() {
    axios.defaults.headers.post['Content-Type'] =
      'application/json;charset=utf-8';
    axios.defaults.headers.post['Access-Control-Allow-Origin'] = '*';
    const arr = [...Array(9)].map((_, i) => i);
    arr.map((_) =>
      axios
        .get(`https://dog.ceo/api/breeds/image/random`)
        .then((res) => {
          this.setState({
            // setState is not function?
            images: this.state.images.concat(res.data.message),
          });
        })
        .catch((error) => console.log(error))
    );
  }
  render() {
    return (
      <div className="article-editor__container--img">
        {this.state.images.map((v, i) => {
          return (
            <img
              key={`${i}_${v.toString()}`}
              src={Spinner}
              id={`img-list-num-${i}`}
              className="article-editor__img--selector"
              onClick={this.props.linkInput}
              onLoad={() => {
                const getId = document.getElementById(`img-list-num-${i}`);
                getId.setAttribute('src', v);
              }}
            />
          );
        })}
      </div>
    );
  }
}

export default Image;
