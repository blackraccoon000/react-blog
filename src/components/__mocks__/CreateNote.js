import React from "react"
const mock = jest.fn().mockImplementation(() => (<div>CreateNote Use ManualMock</div>))
export default mock