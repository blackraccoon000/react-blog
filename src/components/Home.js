import React from "react"
import { Link } from "react-router-dom"
import { connect } from "react-redux"
import ArticleNote from "./ArticleNote"
import getVisibleBlogData from "../selectors/getVisibleBlogData"

const Home = (props) => {
  return (
    <div className="content-container">
      {
        props.blogData.map(note => {
          return props.isAuthenticated ?
          <Link key={`link_${note.id}`} to={`/edit/${note.id}`}>
            <ArticleNote key={note.id} {...note} />
          </Link>
          :
          <ArticleNote key={note.id} {...note} />
        })
      }
    </div>
  )
}

const mapStateToProps = (state)=>{
  return {
    blogData: getVisibleBlogData(state.blogData,state.filters),
    isAuthenticated: !!state.auth.uid
  }
}

export default connect(mapStateToProps)(Home)
export { Home }