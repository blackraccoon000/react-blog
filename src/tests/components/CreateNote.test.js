import React from "react"
import { render, fireEvent, screen, cleanup } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import { CreateNote } from "../../components/CreateNote"

let onChange, onSubmit, wrapper, match, findData

beforeEach(()=>{
  onChange = jest.fn()
  onSubmit = jest.fn()
  match = { params:{ id: undefined }}
  findData = { title:"", note:"" }
  wrapper = render(<CreateNote
    onChange = {onChange}
    onSubmit = {onSubmit}
    match = {match}
    findData = {findData}
  />)
})

it("CreateNote Render", () => {
  expect(wrapper.container).toMatchSnapshot()
})

it("Generate title", () => {
  userEvent.type(wrapper.getByTestId('title'), 'Hello,World!')
  expect(wrapper.getByTestId('title')).toHaveValue('Hello,World!')
})

it("Generate note", () => {
  userEvent.type(wrapper.getByTestId('note'), 'Hello,{enter}World!')
  expect(wrapper.getByTestId('note')).toHaveValue('Hello,\nWorld!')
})