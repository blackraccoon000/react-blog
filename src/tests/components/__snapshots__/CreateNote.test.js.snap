// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateNote Render 1`] = `
<div>
  <div
    class="content-container"
  >
    <h1>
      This is Create Note Page
    </h1>
    
    <div>
      <form>
        <input
          class="createNote-title"
          data-testid="title"
          placeholder="title"
          type="text"
          value=""
        />
        <textarea
          class="createNote-textarea"
          data-testid="note"
          placeholder="Add a note"
        />
        <button>
          save
        </button>
      </form>
    </div>
  </div>
</div>
`;
